music_assistant/controllers/metadata.py

```diff
@@ -261,21 +264,38 @@ async def update_metadata(
     ) -> MediaItemType:
         """Get/update extra/enhanced metadata for/on given MediaItem."""
         if isinstance(item, str):
-            item = await self.mass.music.get_item_by_uri(item)
+            retrieved_item = await self.mass.music.get_item_by_uri(item)
+            if not hasattr(retrieved_item, "provider") or not hasattr(retrieved_item, "media_type"):
+                raise RuntimeError("Retrieved item is not a valid MediaItemType")
```

Comment:
this is not even possible - I would leave this out

Reply:
Apparently it could be a BrowseFolder so I have made some changes which I think is neater.